aopy/preproc/wrappers.py

```diff
@@ -346,18 +351,104 @@ def proc_broadband(data_dir, files, result_dir, result_filename, overwrite=False
+    '''
+    Process ap data
+        neuropixels: filter data between 300-5k, downsample to 10k, synchronize timestamps with ecube. Drift correction is applied
+    Saves ap data into the HDF datasets:
+        ap_data (nt, nch)
+        ap_metadata (dict)
+    
+    Args:
+        data_dir (str): where the data files are located
+        files (dict): dictionary of filenames indexed by system
+        result_dir (str): where to store the processed result
+        result_filename (str): where to store the processed result
+        kilosort_dir (str): data folder where preprocessed data of kilosort is saved
+        overwrite (bool, optional): whether to remove existing processed files if they exist
+        max_memory_gb (float, optional): max memory used to load binary data at one time
+        filter_kwargs (dict, optional): keyword arguments to pass to :func:`aopy.precondition.filter_lfp`
+    '''  
+    # Check if a processed file already exists
+    filepath = os.path.join(result_dir, result_filename)
+    if not overwrite and os.path.exists(filepath):
+        contents = aodata.get_hdf_dictionary(result_dir, result_filename)
+        if "ap_data" in contents:
+            print("File {} already preprocessed, doing nothing.".format(result_filename))
+            return
+    elif os.path.exists(filepath):
+        os.remove(filepath)
+
+    # Preprocess neural data into lfp   
+    if 'neuropixels' in files:
+        np_recorddir = files['neuropixels']
+        ecube_files = files['ecube']
+        datatype = 'ap'
+        
+        nport = len(glob.glob(os.path.join(data_dir, f'{np_recorddir}*/**/continuous/*AP'),recursive=True))
+        
+        for iport in range(nport):
+            iport += 1
+            _, ap_metadata = proc_neuropixel_ts(data_dir, np_recorddir, ecube_files, kilosort_dir, datatype, iport, filepath,
+                                                max_memory_gb = max_memory_gb)
+            aodata.save_hdf(result_dir, result_filename, ap_metadata, f'drive{iport}/ap_metadata', append=True)
+
+def proc_lfp(data_dir, files, result_dir, result_filename, kilosort_dir=None, overwrite=False, max_memory_gb=1., filter_kwargs={}):
     '''
     Process lfp data:
         Loads 'broadband' hdf data or 'ecube' headstage data and metadata
+        neuropixels: low-pass filter data with 500Hz, downsample to 1k, synchronize timestamps with ecube. Destriping is applied
     Saves lfp data into the HDF datasets:
         lfp_data (nt, nch)
         lfp_metadata (dict)
     
     Args:
         data_dir (str): where the data files are located
         files (dict): dictionary of filenames indexed by system
+        result_dir (str): where to store the processed result
         result_filename (str): where to store the processed result
+        kilosort_dir (str): data folder where preprocessed data of kilosort is saved
```

Comment:
can you explain to me what the difference between the kilosort_dir and the data_dir are?

Reply:
data_dir is `/data/raw`
kilosort_dir is `/data/preprocessed/kilosort`. Output by kilosort is saved in this folder.

Reply:
i ask about updating `proc_single` because the arguments to all the other `proc` functions is fixed and it doesn't include `kilosort_dir`

Reply:
yeah you're right. let's keep it separate!